crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1520,6 +1504,48 @@ impl From<FString> for Expr {
     }
 }
 
+/// A newtype wrapper around a list of [`FStringElement`].
+#[derive(Clone, Default, PartialEq)]
+pub struct FStringElements(Vec<FStringElement>);
+
+impl FStringElements {
+    /// Returns an iterator over all the [`FStringLiteralElement`] nodes contained in this f-string.
+    pub fn literals(&self) -> impl Iterator<Item = &FStringLiteralElement> {
+        self.iter().filter_map(|element| element.as_literal())
+    }
+
+    /// Returns an iterator over all the [`FStringExpressionElement`] nodes contained in this f-string.
+    pub fn expressions(&self) -> impl Iterator<Item = &FStringExpressionElement> {
+        self.iter().filter_map(|element| element.as_expression())
+    }
+}
+
+impl From<Vec<FStringElement>> for FStringElements {
+    fn from(elements: Vec<FStringElement>) -> Self {
+        FStringElements(elements)
+    }
+}
+
+impl Deref for FStringElements {
+    type Target = Vec<FStringElement>;
```

Comment:
Nit: Should this be dereferencing to `[FStringElement]` or are there cases where we need a Vec?

Reply:
I don't think so. I think I wanted it to be `[FStringElement]`. I'll change it.